Makefile

```
@@ -656,3 +656,165 @@ endif
+# - Start an Ollama container if not already running
+# - Pull the model if not already cached
+# - Run the code review
+# - Stop and remove the container (but preserve the .ollama cache)
+# 
+# Usage:
+#   make ai-review-gptme-ollama                    # Uses default model (llama3.2:1b)
+#   make ai-review-gptme-ollama OLLAMA_MODEL=phi3:mini    # Uses specified model
+# 
+# Available models (examples):
+#   Small models (< 2GB memory):
+#     - llama3.2:1b (default)
+#     - phi3:mini
+#     - tinyllama
+#   
+#   Medium models (4-8GB memory):
+#     - llama3.2:3b
+#     - gemma2:2b
+#     - gemma3n:e4b (requires ~7GB)
+#     - gemma3n:e2b
+#   
+#   Larger models (8GB+ memory):
+#     - llama3.1:8b
+#     - mistral
+#     - gemma3:12b (11GB)
+
+# suggestions: try gemma3:12b, then gemma3n:e4b, then gemma3n:e2b in order of decreasing memory requirements
+# Default Ollama model (using a smaller model that requires less memory)
+OLLAMA_MODEL ?= gemma3:12b
+# will require at least this much free mem in your machine or podman machine (non-linux)
+OLLAMA_MEMORY ?= 11
+
+# This target reviews staged changes using gptme with Ollama backend
+# Prerequisites:
+#   - gptme installed (pip install gptme)
+#   - Podman installed and running
+# 
+# This target will:
+#   - Create a local .ollama directory for caching models between runs
+#   - Start an Ollama container if not already running
+#   - Pull the model if not already cached
+#   - Run the code review with gptme
+#   - Stop and remove the container (but preserve the .ollama cache)
+#
+# This version enables tools for enhanced review:
+#   - read: Read local files for context (always enabled)
+#   - browser: Browse documentation and references (only if lynx is installed)
+#
+# Usage:
+#   make ai-review-gptme-ollama                           # Uses default Ollama model
+#   make ai-review-gptme-ollama GPTME_OLLAMA_MODEL=phi3   # Uses specific model
+.PHONY: ai-review-gptme-ollama
+ai-review-gptme-ollama: TOOLS = $(shell command -v lynx >/dev/null 2>&1 && echo "read,browser" || echo "read")
+ai-review-gptme-ollama: gptme ## Review staged git changes using gptme with local Ollama models (auto-manages Ollama container)
+	@if [ -z "$$(git diff --cached --name-only)" ]; then \
+		echo "No staged changes to review."; \
+		echo "Please stage your changes first with 'git add <files>'"; \
+		echo "Run 'git status' to see which files are staged."; \
+		exit 0; \
+	fi
+	@# gptme is installed as a dependency, no need to check
+	@# Check if Ollama is already available (either as existing container or local service)
+	@if curl -s http://localhost:11434/api/tags >/dev/null 2>&1; then \
+		echo "Ollama is already running on port 11434"; \
+		OLLAMA_EXTERNAL=1; \
+	else \
+		OLLAMA_EXTERNAL=0; \
+		echo "Ollama not detected, starting container..."; \
+		mkdir -p .ollama; \
+		if ! podman ps | grep -q ollama; then \
+			podman run -d \
+				-v $$(pwd)/.ollama:/root/.ollama \
```

```suggestion
				-v $$(pwd)/.ollama:/root/.ollama:Z \
```
This will solve the permission denied issue for the ssh key